core/lib/multivm/src/versions/vm_refunds_enhancement/vm.rs

```diff
@@ -162,10 +160,7 @@ impl<S: WriteStorage> VmInterfaceHistoryEnabled for Vm<S, HistoryEnabled> {
         self.rollback_to_snapshot(snapshot);
     }
 
-    /// Pop the latest snapshot from the memory and destroy it
     fn pop_snapshot_no_rollback(&mut self) {
-        self.snapshots
-            .pop()
-            .expect("Snapshot should be created before rolling it back");
+        self.snapshots.pop();
```

Comment:
When is it necessary to pop a nonexistent snapshot?

Reply:
It happens [here](https://github.com/matter-labs/zksync-era/blob/b51e26d1a4da11bb244e4c31da4609d13fad0839/core/node/state_keeper/src/batch_executor/main_executor.rs#L172) on the first transaction, or after the previous transaction was rolled back.

Reply:
I think you could just always pop the snapshot if it isn't rolled back in the same place where you may roll it back.